astromodels/functions/functions_2D.py

```diff
@@ -230,4 +231,100 @@ def get_boundaries(self):
 
                 max_lon = max_lon - 360.
 
-        return (min_lon, max_lon), (min_lat, max_lat)
\ No newline at end of file
+        return (min_lon, max_lon), (min_lat, max_lat)
+
+class SpatialTemplate_2D(Function2D):
+    r"""
+        description :
+        
+            User input Spatial Template.  Expected to be normalized to 1/sr
+        
+        latex : $ hi $
+        
+        parameters :
+        
+            K :
+        
+                desc : normalization
+                initial value : 1
+                fix : yes
+        
+        """
+    
+    __metaclass__ = FunctionMeta
+    
+    def _set_units(self, x_unit, y_unit, z_unit):
+        
+        self.K.unit = z_unit
+    
+    # This is optional, and it is only needed if we need more setup after the
+    # constructor provided by the meta class
+    
+    def _setup(self):
+        
+        self._frame = ICRS()
+    
+    def set_file(self,fitsfile,ihdu=0):
+        
+        f = fits.open(fitsfile)
+        self.refXpix = f[ihdu].header['CRPIX1']
```

Comment:
I am going to fix this after the merge, but for future reference,  according to our coding standards, attributes you are not exposing to the public API should called with a name starting with "_".